src/main/kotlin/org/neo4j/graphql/Predicates.kt

```diff
@@ -107,67 +134,45 @@ data class RelationPredicate(val fieldName: String, val op: Operators, val value
+    NEW("_not_ends_with", "ENDS WITH", { lhs, rhs -> lhs.endsWith(rhs).not() }, true),
+    C("_contains", "CONTAINS", { lhs, rhs -> lhs.contains(rhs) }),
+    SW("_starts_with", "STARTS WITH", { lhs, rhs -> lhs.startsWith(rhs) }),
+    EW("_ends_with", "ENDS WITH", { lhs, rhs -> lhs.endsWith(rhs) });
 
     val list = op == "IN"
 
     companion object {
-        private val ops = enumValues<Operators>().sortedWith(Comparator.comparingInt { it.suffix.length }).reversed()
-        val allNames = ops.map { it.suffix }
-        val allOps = ops.map { it.op }
-
-        fun resolve(field: String, value: Any?): Pair<String, Operators> {
-            val fieldOperator = ops.find { field.endsWith("_" + it.suffix) }
-            val unaryOperator = if (value is UnaryOperator) unaryOperatorOf(field, value) else EQ
-            val op = fieldOperator ?: unaryOperator
-            val name = if (op.suffix.isEmpty()) field else field.substring(0, field.length - op.suffix.length - 1)
-            return name to op
-        }
 
-        private fun unaryOperatorOf(field: String, value: Any?): Operators =
-                when (value) {
-                    is IsNullOperator -> if (field.endsWith("_not")) IS_NOT_NULL else IS_NULL
-                    else -> throw IllegalArgumentException("Unknown unary operator $value")
-                }
-
-        fun forType(type: GraphQLInputType): List<Operators> =
-                if (type == Scalars.GraphQLBoolean) listOf(EQ, NEQ)
-                else if (type is GraphQLEnumType || type is GraphQLObjectType || type is GraphQLTypeReference) listOf(EQ, NEQ, IN, NIN)
-                else listOf(EQ, NEQ, IN, NIN, LT, LTE, GT, GTE) +
-                        if (type == Scalars.GraphQLString || type == Scalars.GraphQLID) listOf(C, NC, SW, NSW, EW, NEW) else emptyList()
+        fun resolve(queriedField: String, field: String, value: Any?): FieldOperator? {
+            if (value == null) {
+                return listOf(IS_NULL, IS_NOT_NULL).find { queriedField == field + it.suffix } ?: return null
+            }
+            val fieldOperator = enumValues<FieldOperator>()
+                .filterNot { it == IS_NULL || it == IS_NOT_NULL }
+                .find { queriedField == field + it.suffix } ?: return null
+            val op = fieldOperator
+            return op
+        }
 
-        fun forType(type: GraphQLType): List<Operators> =
+        fun forType(type: GraphQLType): List<FieldOperator> =
                 when {
                     type == Scalars.GraphQLBoolean -> listOf(EQ, NEQ)
                     type.isNeo4jType() -> listOf(EQ, NEQ, IN, NIN)
-                    type is GraphQLFieldsContainer || type is GraphQLInputObjectType -> listOf(IS_NULL, IS_NOT_NULL, SOME, NONE, SINGLE)
+                    type is GraphQLFieldsContainer || type is GraphQLInputObjectType -> throw IllegalArgumentException("This operators are not for relations, use the RelationOperator instead")
```

Comment:
what should user do with this error message?

Reply:
This is nothing the user will see, it is an error in the implementation